test/benchmark/PhpUnitTestCaseBench.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Roave\BetterReflectionBenchmark;
+
+use PHPUnit\Framework\TestCase;
+use Roave\BetterReflection\BetterReflection;
+use Roave\BetterReflection\Reflector\ClassReflector;
+use Roave\BetterReflection\Reflection\ReflectionMethod;
+use Roave\BetterReflection\Reflection\ReflectionParameter;
+
+/**
+ * @Iterations(10)
+ */
+class PhpUnitTestCaseBench
+{
+    /**
+     * @var ClassReflector
+     */
+    private $reflector;
+
+    public function __construct()
+    {
+        $reflection = new BetterReflection();
+        $this->reflector = $reflection->classReflector();
+    }
+
+    /**
+     * @Subject()
+     */
+    public function reflect_phpunit_test_case()
+    {
+        $reflection = $this->reflector->reflect(TestCase::class);
```

Comment:
Any way to isolate parsing operations from reflection ones?

Reply:
I think you would need a profiler for that, there is also the [XDebug](http://phpbench.readthedocs.io/en/latest/extensions/xdebug.html#function-tracing) extension which can help.

Reply:
I was asking to split the `$this->reflector->reflect(TestCase::class);` call into a preparation step, while the actual looping over methods/parameters should happen in here

Reply:
Well, you are still going to get contamination think? - `TestCase` extends `Assert` - not sure if it will trigger any more loading.

Reply:
It will, but what bit are you benchmarking? 

```php
$reflection = $this->reflector->reflect(TestCase::class);
```

or 

```php
$method->hasReturnType() ? $method->getReturnType()->__toString() : null;
```

or

```php
$parameter->hasType() ? $parameter->getType()->__toString() : null;
```

Reply:
The context is that I encountered the performance issue wjen writing an auto-complete tool, and the performance was particularly bad on the `TestCase`. The tool iterates over methods, parameters, properties, etc. So this is kind-of reproducing that real-world scenario.

Happy to add smaller methods, but I think having the scenario is good (it should probably iterate over properties too). (

btw I already wrote similar tests for [here](https://github.com/phpactor/worse-reflection/tree/master/tests/Benchmarks).

Reply:
> Happy to add smaller methods, but I think having the scenario is good (it should probably iterate over properties too).

I would really want to isolate the performance issues here, rather than grouping them.
Sorry if I sound annoying :-\